common/src/main/java/io/pravega/common/util/Retry.java

```diff
@@ -202,11 +202,14 @@ private RetryAndThrowBase(Class<ThrowsT> throwType, RetryWithBackoff params) {
                     }
                 }
 
-                final long sleepFor = delay;
-                Exceptions.handleInterrupted(() -> Thread.sleep(sleepFor));
+                if (attemptNumber < params.attempts) {
```

Comment:
runAsync works fine. With this change the timing behavior of .run and runAsync will be the same.